tests/suites/hup/tests/rollback-altboot.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 balena
+ *
+ * @license Apache-2.0
+ */
+
+'use strict';
+
+module.exports = {
+  title: 'Rollback altboot (broken init) test',
+  run: async function(test) {
+    await this.context.get().hup.initDUT(
+      this, test, this.context.get().link
+    );
+
+    const before = await this.context.get().worker.getOSVersion(this.context.get().link);
+    test.comment(`VERSION (before): ${before}`);
+
+    await this.context.get().hup.doHUP(this, test, 'image', this.context.get().hup.payload, this.context.get().link);
+
+    // there are no timers/counts to change for rollback-altboot, but rollback-health also gets called
+    // so we will reduce the timeouts used there as it comes into play
+  
+    // reduce number of failures needed to trigger rollback
+    test.comment(`Reducing amount of time needed for rollback-health to fail...`)
+    await this.context.get().worker.executeCommandInHostOS(
+      `sed -i -e "s/COUNT=.*/COUNT=3/g" -e "s/TIMEOUT=.*/TIMEOUT=30/g" $(find /mnt/sysroot/inactive/ | grep "bin/rollback-health")`,
+      this.context.get().link,
+    );
+
+    // break init
+    test.comment(`Breaking init to trigger rollback-altboot...`)
+    await this.context.get().worker.executeCommandInHostOS(
+      `rm /mnt/sysroot/inactive/current/boot/init`,
+      this.context.get().link,
+    );
+
+    await this.context.get().worker.rebootDut(this.context.get().link);
+
+    // on a passed HUP, rollback-health will clean up the breadcrumbs and run indefinitely
+    // on a failed HUP, rollback-health will clean up the breadcrumbs and reboot again
+    // so by waiting for a true 'inactive' state we can trust that the reboots are done
+    test.comment(`Waiting for rollback-health service to be inactive...`);
+    await this.context.get().utils.waitUntil(async () => {
+      return (
+        (await this.context.get().worker.executeCommandInHostOS(
+          `systemctl is-active rollback-health.service`,
+          this.context.get().link,
+        )) === 'inactive'
+      );
+    }, false);
```

Comment:
I'm using a waitUntil block to wait until healthchecks have finished AND a second reboot happens without us initiating one. However, this seems to get the waitUntil block stuck forever:
```
[2021-07-15T20:52:38.963Z][06c77b8-hup] Rebooting the DUT
[2021-07-15T20:53:39.947Z][06c77b8-hup] DUT has rebooted & is back online
[2021-07-15T20:53:39.952Z][06c77b8-hup]             # Waiting for rollback-health service to be inactive...
```
Is there a better way to accomplish this? Do we have a helper that waits for a reboot without initiating one?

Reply:
Not at the moment though. Since the need didn't arise for it but writing a custom function right above the test to achieve would be easier. 

You can remove the part where we actually reboot from the rebootDUT helper and replace that with the assertion condition of checking if the temporary file we created (before reboot) is still there or not. 

If a reboot has happened, then the file will not be there and you can have a waitUntil loop checking if the file has been removed yet. That should work out.

Reply:
I will look into something like that, it seems this test is broken either way. I'll fix it up and see why it never finds the inactive/active services.

Weird that it took 41min to fail though, I would have expected 10 min or so? We still must be missing something here?